util/util.cpp

```diff
@@ -15,12 +15,16 @@
 #include "sensor/sensor.hpp"
 #include "dbus/dbus.hpp"
 
-// Enables logging of margin decisions
-static constexpr bool DEBUG = false;
 
 // Enables single-point-of-failure behavior
 bool spofEnabled = false;
 
+// Enables logging of margin decisions
+bool debugEnabled = false;
+
+// Flag for sensor fail
+bool errorEncountered = false;
```

Comment:
It isn't strictly necessary to make errorEncountered a global. A return of NaN from getSensorDbusTemp() will set errorEncountered locally, see the std::isfinite() check on line 402.

Reply:
Oh, it's my mistake, I will fix it.
But I have a question for errEncountered, if not found the sensor's service path(line 63), is it need to make fanfail?
For example, if user's memory slot is not full and all memory is in a zone in config.json, it will let fan to fanfail mode in current code.

Reply:
Good question. That is a policy decision. The correct answer depends on the user's decision. In our application, yes, all sensors must be present, and it is correct to declare an error condition if a sensor is missing. However, another user could feel differently. Do you think we should make this a configurable option?

Reply:
I think we can make a option for this situation that maybe useful for other user or in development phase.
I will implement this. Thanks.

Reply:
Done.